pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.js

```diff
@@ -29,20 +30,27 @@ router.get('/', function (req, res, next) {
   setFilenames(res.locals);
   async.series(
     [
-      function (callback) {
-        var params = { assessment_id: res.locals.assessment.id };
-        sqldb.queryOneRow(sql.assessment_stats_last_updated, params, function (err, result) {
-          if (ERR(err, callback)) return;
-          res.locals.stats_last_updated = result.rows[0].stats_last_updated;
-          callback(null);
+      async () => {
+        // make sure statistics are up to date
+        await assessment.updateAssessmentStatistics(res.locals.assessment.id);
+      },
+      async () => {
+        // re-fetch assessment to get updated statistics
+        const result = await sqldb.queryOneRowAsync(sql.select_assessment, {
+          assessment_id: res.locals.assessment.id,
         });
+        res.locals.assessment = result.rows[0].assessment;
       },
       function (callback) {
-        debug('query assessment_stats');
+        // Fetch assessments.stats_last_updated (the time when we last updated
+        // the _question_ statistics for this assessment). Note that this is
+        // different to assessments.statistics_last_updated_at (the time we last
+        // updated the assessment instance statistics stored in the assessments
+        // row itself).
```

Comment:
The preexisting `assessments.stats_last_updated` column is very confusingly named, now that we have added `assessments.statistics_last_updated_at`. Ideally we'd rename `stats_last_updated` to `question_stats_last_updated`, but column renames are a bit unpleasant to manage and I want to eliminate `stats_last_updated` by rewriting question statistic caching using a similar scheme to the current PR. However I'll defer that to a future PR and so I'm just leaving `stats_last_updated` as it is for the moment. Is there somewhere else that we should add a comment about this confusing naming situation?

Reply:
Good point. https://github.com/PrairieLearn/PrairieLearn/issues/6743